src/components/icon/get-icon-props.ts

```diff
@@ -0,0 +1,117 @@
+import { Icon } from '../../interface';
+
+/**
+ * Get the icon name.
+ * This helps in setting the right icon, both if the consumer only types `icon: string`,
```

Comment:
now after this PR, consumers can either write `icon: 'icon-name'` or `icon: {name: 'icon-name'}`.

However the only component that only works by writing `icon: {name: 'icon-name'}` is `limel-breadcrumbs`. I wonder if we should do so that both ways work for breadcrumbs as well? 🤔 otherwise, it'll be weird.

Reply:
I buy that. But from a consumer's perspective it also felt convenient (in a smart way) not to _have to_ specify `icon` and then `name`, if all you need is the `name` (you're not interested in `title`, `color`, etc…). Also in the docs, we don't say that this -> `icon: {name: 'icon-name'}` is also deprecated.

Reply:
Ah! I think I misunderstood. Yeah, I think the property `icon` should be of type `string | Icon` everywhere, and if it's just a string, we use that for the name of the icon to show. Absolutely 👍

I thought you were asking whether we should implement `iconColor` etc on breadcrumbs. Sorry, I clearly wasn't paying attention to what I was reading 🙈

Reply:
np!

I added a commit here for ti [`6841531` (#2542)](https://github.com/Lundalogik/lime-elements/pull/2542/commits/68415317595d5c0b8ca057668e8dcf2387403b02)

@BjornOstberg I hope this does not override [your recent fix for breadcrumbs](https://github.com/Lundalogik/lime-elements/pull/2550)?

Reply:
I think it will work the same, so it shouldn't be a problem.